src/utils/work/seekWorkResponse.js

```diff
@@ -146,30 +134,43 @@ const dispatchWorkRequest = async (
   });
 
   const responsePromise = new Promise((resolve, reject) => {
-    io.on(`WorkResponse-${ETag}`, (res) => {
-      const { response } = res;
-
-      if (response.error) {
-        const { errorCode, userMessage } = response;
-        console.error(errorCode, userMessage);
-
-        return reject(
-          new WorkResponseError(errorCode, userMessage, request),
-        );
+    io.on(`WorkResponse-${ETag}`, async (res) => {
+      // If type is object, then we received a notification with a signedUrl
+      // now we need to fetch the actual result from s3
```

Comment:
This takes place in 2 cases:
- The work result is too big to be returned over sockets (it's going to be slow) so we return a signedUrl and the UI needs to fetch it.
- The work result is cached, so the UI can fetch it directly from S3